tasks/main.yml

```diff
@@ -1,27 +1,106 @@
   package:
     name: "{{ __bootloader_packages }}"
     state: present
 
-- name: Ensure required services are enabled and started
-  service:
-    name: "{{ item }}"
-    state: started
-    enabled: true
-  loop: "{{ __bootloader_services }}"
-
-- name: Generate /etc/{{ __bootloader_foo_config }}
-  template:
-    src: "{{ __bootloader_foo_config }}.j2"
-    dest: /etc/{{ __bootloader_foo_config }}
-    backup: true
-    mode: 0400
-  notify: Bootloader handler to restart services
+- name: Remove the existing boot settings
+  when: bootloader_settings |
+    selectattr('previous', 'defined') |
+    selectattr('previous', 'match', '^replaced$') |
+    list | length > 0
+  block:
+    - name: Get existing boot settings
+      shell: >-
+        set -o pipefail;
+        grubby --info=DEFAULT | grep '^args="' | sed 's/^args=//'
+      changed_when: false
+      register: __bootloader_args
+
+    - name: Remove all boot settings
+      command: >-
+        grubby
+        --update-kernel=DEFAULT
+        --remove-args={{ __bootloader_args.stdout }}
+      changed_when: true
+      # length > 2 for two quotes
+      when: __bootloader_args.stdout | length > 2
+      notify:
+        - Write boot loader configuration
+        - Reboot system
+
+- name: Modify boot settings
+  include_tasks: modify_settings.yml
+  loop: "{{ bootloader_settings }}"
+
+# A workaround for https://bugzilla.redhat.com/show_bug.cgi?id=1152027
+- name: Fix default kernel boot parameters (EL 7)
+  shell: |-
+    set -o pipefail
+    eval $(grubby --info=DEFAULT | awk '/^args/ {print $0}')
+    sed -i -e "s,^GRUB_CMDLINE_LINUX=.*,GRUB_CMDLINE_LINUX=\"$args\","
+    {{ __bootloader_default_grub }}
+  changed_when: true
+  when:
+    - ansible_distribution in ['CentOS', 'RedHat']
+    - ansible_facts.distribution_major_version is version('7', '=')
+    - __bootloader_conf_boot_params is changed
+
+- name: Update boot loader timeout configuration
+  lineinfile:
+    path: "{{ __bootloader_default_grub }}"
+    regexp: '^GRUB_TIMEOUT=.*'
+    line: 'GRUB_TIMEOUT={{ bootloader_timeout }}'
+  notify: Write boot loader configuration
+
+- name: Determine platform type
+  stat:
+    path: /sys/firmware/efi
+  register: efi_dir
+
+- name: Set boot loader configuration files
+  vars:
+    efi_path: "{{
+      ansible_distribution in ['CentOS', 'RedHat'] and
+      ansible_facts.distribution_major_version | int < 9 and
+      efi_dir.stat.exists }}"
+  set_fact:
+    __bootloader_grub_conf: >-
+      {{ efi_path | ternary('/boot/efi/EFI/redhat/grub.cfg',
+      '/boot/grub2/grub.cfg') }}
+    __bootloader_user_conf: >-
+      {{ efi_path | ternary('/boot/efi/EFI/redhat/user.cfg',
+      '/boot/grub2/user.cfg') }}
+
+- name: Update boot loader password
+  when: bootloader_password is not none
+  block:
+    - name: Generate boot loader password
+      shell: >-
+        set -euo pipefail;
+        ( echo {{ bootloader_password }} ; echo {{ bootloader_password }} )
```

Comment:
```suggestion
        ( echo {{ bootloader_password | quote }} ; echo {{ bootloader_password | quote }} )
```

Reply:
and maybe there is a more secure way to do this

Reply:
Unfortunately, no. The usual way to do this is by running `grub2-setpassword`, but it does not allow inputtng via echo:
```
# ( echo "12345" ; echo "12345" ) | grub2-setpassword
stty: standard input: Inappropriate ioctl for device
```
There is no way to input other than `echo`, no env var for this or so.